src/include/storage/store/struct_column_chunk.h

```diff
@@ -7,13 +7,16 @@ namespace storage {
 
 class StructColumnChunk : public ColumnChunk {
 public:
-    StructColumnChunk(common::LogicalType dataType, uint64_t capacity, bool enableCompression);
+    StructColumnChunk(
+        common::LogicalType dataType, uint64_t capacity, bool enableCompression, bool needFinalize);
```

Comment:
I think it's better for us to wrap logicalType inside unique_ptr to avoid unintentional copy.